src/dune_rules/lib.ml

```diff
@@ -1116,7 +1166,92 @@ end = struct
   ;;
 
   let instantiate db name info ~hidden = (Lazy.force db.instantiate) name info ~hidden
-  let find_internal db (name : Lib_name.t) = resolve_name db name
+
+  let resolve_hidden db ~info hidden =
+    let open Memo.O in
+    (match db.parent with
+     | None -> Memo.return Status.Not_found
```

Comment:
~Shouldn't we return `Hidden` here?~ Edit: i see now that this is how existing code worked 🤷